lib/utils/uacc.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+// #include <stdlib.h>
+// #include "uacc.h"
+import "C"
+
+import (
+	"encoding/binary"
+	"errors"
+	"net"
+	"sync"
+	"unsafe"
+)
+
+// This module concerns itself with updating the user account database and log on nodes
+// that a client connects to with an interactive session.
+
+// Due to thread safety design in glibc we must serialize all access to the accounting database.
+var accountDb sync.Mutex
```

Comment:
Why does it have to be package-global? Can it be encapsulated in a type and a value of that type could be created for a server instance?

Reply:
It has to be package global because this state is managed globally within glibc. Can't do much about it :/